crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -147,25 +147,27 @@ async fn create(
             )?;
         }
     }
-
-    let upsert = diesel::insert_into(default_configs)
-        .values(&default_config)
-        .on_conflict(db::schema::default_configs::key)
-        .do_update()
-        .set(&default_config)
-        .execute(&mut conn);
-
-    match upsert {
-        Ok(_) => Ok(HttpResponse::Ok().json(json!({
-            "message": "DefaultConfig created/updated successfully."
-        }))),
-        Err(e) => {
-            log::info!("DefaultConfig creation failed with error: {e}");
-            Err(unexpected_error!(
-                "Something went wrong, failed to create DefaultConfig"
-            ))
-        }
-    }
+    conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+        let upsert = diesel::insert_into(default_configs)
+            .values(&default_config)
+            .on_conflict(db::schema::default_configs::key)
+            .do_update()
+            .set(&default_config)
+            .execute(transaction_conn);
+        add_config_version(&state, req.tags, transaction_conn)?;
```

Comment:
@pratikmishra356 , do we have to fail a request just because we weren't able to create a snapshot.
What I think, the user never intended to make a snapshot, he just wanted to create a default_config/context/....., creating a snapshot should be our separate task, I feel we really shouldn't fail the endpoints when we fail to create a snapshot

@knutties @Datron , what are you thoughts on this?

Reply:
since currently we dont have any process/event based tracking mechanisms to do these operations in async , we might lose track of updates on configs thats why we followed this behavior , 
we assume every updates on default_config and contexts is crucial which results in changing the configs , and since now we are relying only on config_version table for serving configs ( i.e no more db fetch fetch from multiple tables ) it is important to have this both operation in sync